confdgnmi/testtool/OpenConfig/01__interfaces.robot

```diff
@@ -0,0 +1,195 @@
+    [Documentation]    Verify that list can be iterated with separate prefix/path parameters.
+    [Tags]  prefix  list
+
+Read "/interfaces/interface[]/name" key leaf from existing entry
+    [Documentation]    Verify that key value can be read using Get request with path parameter.
+    [Tags]  key  list
+    ${response}=  Get data from  /interfaces/interface[name=${INTERFACE}]/name
+    Should Be Equal    ${INTERFACE}    TODO
+
+Get "CONFIG" response includes "config" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=CONFIG`` parameter receives
+    ...                OK response that does include the read/write YANG "config" container.
+    [Tags]  config  container
+    Set Datatype To    CONFIG
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check Last Updates Include    config
+
+Get "CONFIG" response does not include "state" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=CONFIG`` parameter receives
+    ...                OK response that does not include the read-only YANG "state" container.
+    [Tags]  config  container  negative
+    Set Datatype To    CONFIG
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates not include    state
+
+Get "OPERATIONAL" response includes "state" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=OPERATIONAL`` parameter receives
+    ...                OK response that does include the read/write YANG "state" container.
+    [Tags]  operational  container
+    Set Datatype To    OPERATIONAL
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates include  state
+
+Get "OPERATIONAL" response does not include "config" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=OPERATIONAL`` parameter receives
+    ...                OK response that does not include the read-only YANG "config" container.
+    [Tags]  operational  container  negative
+    Set Datatype To    OPERATIONAL
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates not include  config
+
+Get "ALL" response includes both "config" and "state" containers
+    [Documentation]    Verify that ``GetRequest`` with ``type=ALL`` parameter receives
+    ...                OK response that does include both read-write "config" and read-only "state" containers.
+    [Tags]  config  operational  container
+    Set Datatype To    ALL
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates include  config
+    Check last updates include  state
+
+Get "config" response includes significant leaves
+    [Documentation]    Verify that ``GetRequest`` with ``type=CONFIG`` parameter receives
+    ...                OK response that does include standardized leaf names/items.
+    [Tags]  config  leaf
+    [Template]    Get interface "config" includes leaf
+    name
+    type
+    mtu
+    loopback-mode
+    description
```

Comment:
I agree that value of this may be questionable depending on testing goals.
I assumed tester wants to check "compliance" with standard YANG, thus included all the leaves.
`[Template]` tests are run as separate "tests" and can succeed/fail independently on each other...
This implies that "failure" for some of the leaves like `description` may be just a warning flag that device deviates from standard YANG. We may need to address our stance on cases like this one. (@waitai ?)

Reply:
If user does not need some leaves he can skip specific tests and/or ignore their failure results. 
If user wants 100% OpenConfig compliance, presence of all the leaves is to be expected.

Failing test case does not mean that device is "bad", it just shows what was tested, and what succeeded/failed.
Test case helps to show what target gNMI server device is capable of, and listing negative but reasonably expectable use-cases is IMHO useful as well.

Making any extra things configurable raises knowledge requirement on the test runner/user. This can be problematic due to forcing user to dig into potentially non-interesting details for every other test-case.

Reply:
There may be default config. But the main question is how you know the chosen leafs are significant?

Reply:
that's up to each test user to either skip or ignore leaves from standard that are not "significant" for their use-case :)